newa/cli.py

```diff
@@ -194,50 +193,99 @@ def cmd_event(ctx: CLIContext, errata_ids: list[str]) -> None:
 def cmd_jira(ctx: CLIContext, issue_config: str) -> None:
     ctx.enter_command('jira')
 
-    # this is here temporarily so we generate fake Jira issue IDs
-    jira_id_gen = itertools.count(start=1)
+    # Get control of Jira.
+    # TODO: Hardcoded project for now.
+    # It will be served from the component-config.yaml later.
+    jira = JiraHandler("BASEQESEC")
 
     for erratum_job in ctx.load_erratum_jobs('event-'):
         # read Jira issue configuration
         config = ErratumConfig.from_yaml_file(Path(os.path.expandvars(issue_config)))
 
-        # TODO: record created/existing issues. Instead of `Any`, maybe something
-        # from the Jira library would be stored. Or just a Jira ticket ID.
-        known_issues: dict[str, Any] = {}
+        # Record created/existing issues.
+        known_issues: dict[str, Issue] = {}
 
         # Iterate over issue actions. Take one, if it's not possible to finish it,
         # put it back at the end of the queue.
         issue_actions = config.issues[:]
 
         while issue_actions:
             action = issue_actions.pop(0)
+            rendered_summary = render_template(action.summary, ERRATUM=erratum_job)
+            rendered_description = render_template(action.description, ERRATUM=erratum_job)
+            rendered_assignee = render_template(action.assignee, ERRATUM=erratum_job)
 
+            # For debugging.
             print(f'* Would create a {action.type.name} issue:')
             print(f'     summary: {action.summary}')
             print(f'     summary: {action.description}')
             print()
 
-            if action.id in known_issues:
-                raise Exception(f'Issue "{action.id}" is already created!')
-
+            # Find existing issues in Jira related to erratum_job and action.id.
+            #
+            # We re looking for Jira issues such that:
+            #
+            # 1) summary matches rendered_summary (e.g. Errata respin 0) and
```

Comment:
Again, probably not something for now but in the future we should probably have functions helping building the search query for every EventType.

Reply:
Btw, let's remove that `'respin 0'` string as it is confusing. At first, we are searching for issues independent on a respin.

Reply:
This should be fixed in jinja template. I'll do that.

Reply:
Fixed in the configuration sample, a bit ugly but should work.